dashboard/DashboardDB.py

```diff
@@ -0,0 +1,344 @@
+        update_time = datetime.now()
+        if row is None:
+            cursor.execute(
+                "INSERT INTO {} VALUES(?, ?, ?, ?)".format(table),
+                (day, 0, total_core_usage, update_time),
+            )
+        else:
+            chours_usage =  total_core_usage - row[1] if row[1] is not None else 0
+            if row[0] is not None:
+                chours_usage = row[0] + chours_usage
+
+            cursor.execute(
+                "UPDATE {} SET CORE_HOURS_USED = ?, TOTAL_CORE_HOURS = ?, \
+                UPDATE_TIME = ? WHERE DAY == ?;".format(table),
+                (chours_usage, total_core_usage, update_time, day),
+            )
+
+        conn.commit()
+        conn.close()
+
+    def get_daily_chours_usage(self, day: Union[date, str], hpc: const.HPC):
+        """Gets the usage and total usage for the specified day"""
+        conn = sql.connect(self.db_file)
+        cursor = conn.cursor()
+
+        day = day.strftime(self.date_format) if type(day) is date else day
+        result = cursor.execute(
+            "SELECT CORE_HOURS_USED, TOTAL_CORE_HOURS FROM {} WHERE DAY == ?".format(
+                self.get_daily_t_name(hpc)
+            ),
+            (day,),
+        ).fetchone()
+
+        conn.close()
+        return result
+
+    def update_squeue(self, squeue_entries: Iterable[SQueueEntry], hpc: const.HPC):
+        """Updates the squeue table with the latest queue status"""
+        conn = sql.connect(self.db_file)
+        cursor = conn.cursor()
+
+        table = self.get_squeue_t_name(hpc)
+
+        # Drop and re-create the table
+        cursor.execute("DROP TABLE IF EXISTS {}".format(table))
+        self._create_queue_table(cursor, hpc)
+
+        update_time = datetime.now()
+        for ix, entry in enumerate(squeue_entries):
+            cursor.execute(
+                "INSERT INTO {} VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)".format(table),
+                (
+                    entry.job_id,
+                    update_time,
+                    entry.username,
+                    entry.status,
+                    entry.name,
+                    entry.run_time,
+                    entry.est_run_time,
+                    entry.nodes,
+                    entry.cores,
+                ),
+            )
+
+        conn.commit()
+        conn.close()
+
+    def get_sqeue_entries(self, hpc: const.HPC):
+        """Gets all squeue entries"""
+        conn = sql.connect(self.db_file)
+        cursor = conn.cursor()
+
+        results = cursor.execute(
+            "SELECT JOB_ID, USERNAME, STATUS, NAME, RUNTIME, \
+            ESTIMATED_TIME, NODES, CORES  FROM {};".format(
+                self.get_squeue_t_name(hpc)
+            )
+        ).fetchall()
+
+        conn.close()
+        return [SQueueEntry(*result) for result in results]
+
+    def _create_queue_table(self, cursor, hpc: const.HPC):
+        # Add latest table
+        cursor.execute(
+            """CREATE TABLE {}(
```

Comment:
CREATE TABLE IF NOT EXISTS 

So it doesn't break if it already exists even though technically it never should.